### DISCUSSION TITLE
Wpf动画里面的绑定问题

### DISCUSSION THREAD

**Discussion by yangf85** [BYJRK/DotNet-Discussions#119] [WPF 开发]
`Trigger Property="IsChecked" Value="True">
                <Trigger.EnterActions>
                    <BeginStoryboard>
                        <Storyboard>
                            <DoubleAnimation
                                Storyboard.TargetName="InnerEllipse"
                                Storyboard.TargetProperty="Width"
                                From="0"
                                To="{Binding ElementName=InnerEllipse, Path=ActualWidth}"
                                Duration="0:0:0.3" />
                            <DoubleAnimation
                                Storyboard.TargetName="InnerEllipse"
                                Storyboard.TargetProperty="Height"
                                From="0"
                                To="{Binding ElementName=InnerEllipse, Path=ActualWidth}"
                                Duration="0:0:0.3" />
                        </Storyboard>
                    </BeginStoryboard>
                </Trigger.EnterActions>
            </Trigger>`

如果对From或者To使用绑定属性就会出现这个问题：InvalidOperationException: 无法冻结此 Storyboard 时间线树供跨线程使用。
只能把这个值写死，或者使用后台代码。
还有其他的方式能解决这个问题吗?

**Comment by BYJRK** [comment]
你可以考虑写一个行为在后台代码中生成动画，然后暴露两个依赖属性用来绑定。生成动画时，从依赖属性获取当前的值

**Reply by yangf85** [reply]
这个Behavior在控件模板里面不好使啊
